ambassador/ambassador/config/resourcefetcher.py

```diff
@@ -687,6 +694,15 @@ def handle_k8s_service(self, k8s_object: AnyDict) -> HandlerResult:
                 'namespace': resource_namespace,
                 'ports': ports
             }
+
+            selector = spec.get('selector', {})
+            for key, value in selector.items():
+                if key == 'service' and value == 'ambassador' and resource_name == 'ambassador':
```

Comment:
@kflynn need to talk to you if you think this is the right way to look for Ambassador's service

Reply:
Uhhhh... I don't have anything better. Is there any way you could imagine allowing the user to force a particular service to be the right one?

Reply:
Glossing over this one thing I can see is that my services are not all called `ambassador` for example I have `ambassador-private` and `ambassador-public` and I don't think this logic would locate the service properly.

I will get it fired up tonight after work and look in more detail, but that was my first observation.

Reply:
@stefansedich you're correct - we ship the ambassador pod with the label `service: ambassador` and that's the selector that ambassador service looks for. But when we do the opposite and look for a service with the selector `service: ambassador` there is ambassador and ambassador-admin, which is why I tied it the name to `ambassador`. But clearly that's not ideal.

Reply:
Does it currently handle multiple ambassador instances with an ambsassador_id set not running in a single namespace?

I have not ran this yet but to me it looks like it might store the last ambassador service it matches in `ambassador_service_raw` and then when processing Ingress they would all get linked to this one, however it might not be the ambassador instance it expects?

Reply:
nope, Ambassador IDs are not handled yet.

Reply:
@containscafeine I see a couple options here, we can do the following:

1. Label the service with a `getambassador.io/ambassador-id` if set
2. In `handle_k8s_service` if we see a service has the `app.kubernetes.io/name=ambassador` label set add it to a new dictionary, something like: `self.ambassador_services[ambassador_id] = x` where ambassador_id comes from either the label or we default to `default`.
3. In `handle_k8s_ingress` attempt to locate the correct service but looking at the `self.ambassador_services` dictionary for the correct instance using either the ambassador_id set in the ingress annotation or default.

Now the second idea after digging into Traefik quickly is to populate the service name into some configuration variable defining the service and just use this to the load the service.

https://github.com/containous/traefik/blob/2471f893e7d8b20cb9ee7051fbc97eaef1b023d7/docs/content/providers/kubernetes-ingress.md#ingressendpoint
https://github.com/helm/charts/blob/4a8d788986ee3700ef5eb4e178ef92fa5d605ca6/stable/traefik/templates/configmap.yaml#L180